extensions/sequence.md

```diff
@@ -0,0 +1,46 @@
+# Sequence
+
+This extension defines two attributes that can be included within a CloudEvent
+to describe the position of an event in the ordered sequence of events produced
+by a unique event source.
+
+The `sequence` attribute represents the value of this event's
+order in the stream of events.  The exact value and meaning of this
+attribute is defined by the `sequenceType` attribute. 
+If the `sequenceType` is missing, or not defined in this specification,
+event consumers will need to have some out-of-band communication with the
+event producer to understand how to interpret the value of the attribute.
+
+## Attributes
+
+### sequence
+* Type: `String`
+* Description: Value expressing the relative order of the event. This enables
+  interpretation of data supercedence.
+* Constraints
+  * REQUIRED
+  * MUST be a non-empty lexicographically-orderable string
+  * RECOMMENDED as monotonically increasing and contiguous
+
+### sequenceType
+* Type: `String`
+* Description: Specifies the semantics of the sequence attribute.
+  See the [SequenceType Values](#sequencetype-values) section for more
+  information.
+* Constraints:
+  * OPTIONAL
+  * If present, MUST be a non-empty string
+
+## SequenceType Values
+
+This specification defines the following values for `sequenceType`.
+Additional values MAY be defined by other specifications.
+
+### Integer
+If the `sequenceType` is set to `Integer`, the `sequence` attribute has
+the following semantics:
+* The values of `sequence` are string-encoded signed 32-bit Integers.
+* The sequence MUST be contiguous, monotonically increasing by `1` and
+  start with a value of `1`.
```

Comment:
Sorry to nit.  It seems like it might read more naturally as
```
The sequence MUST start with a value of `1` and increase by `1` for each 
subsequent value (i.e. be contiguous and monotonically increasing).
```